go-controller/pkg/ovn/controller/admin_network_policy/admin_network_policy.go

```diff
@@ -0,0 +1,664 @@
+		acl := libovsdbutil.BuildANPACL(
+			getANPRuleACLDbIDs(anpName, rule.gressPrefix, fmt.Sprintf("%d", rule.gressIndex), c.controllerName, isBanp),
+			int(rule.priority),
+			match,
+			rule.action,
+			libovsdbutil.ACLDirectionToACLPipeline(libovsdbutil.ACLDirection(rule.gressPrefix)),
+		)
+		acls = append(acls, acl)
+		return acls
+	}
+
+	// we have ports defined in this rule, so its one ACL per port
+	// TODO(tssurya): There are some improvement PRs around this, adapt them in the future iterations
+	for i, port := range rule.ports {
+		l4Match := constructMatchFromPorts(port)
+		match = fmt.Sprintf("%s && %s && %s", l3Match, lportMatch, l4Match)
+		acl := libovsdbutil.BuildANPACL(
+			getANPRuleACLDbIDs(anpName, rule.gressPrefix, fmt.Sprintf("%d.%d", rule.gressIndex, i), c.controllerName, isBanp),
+			int(rule.priority),
+			match,
+			rule.action,
+			libovsdbutil.ACLDirectionToACLPipeline(libovsdbutil.ACLDirection(rule.gressPrefix)),
+		)
+		acls = append(acls, acl)
+	}
+
+	return acls
+}
+
+// constructIPSetOfPeers takes all the peers belonging to each of the ANP rule and initiates the conversion
+// of rule.peer->set of ips. These set of ips are then used to create the address-sets
+func (c *Controller) constructIPSetOfPeers(anp *adminNetworkPolicyState) error {
+	// create address-set
+	// TODO (tssurya): Revisit this logic to see if its better to do one address-set per peer
+	// and join them with OR if that is more perf efficient. Had briefly discussed this OVN team
+	// We are not yet clear which is better since both have advantages and disadvantages.
+	// Decide this after doing some scale runs.
+	// TODO(tssurya): Use address-sets of namespaces whereever it makes sense; specially if namespace is already created
+	var err error
+	for _, ingressRule := range anp.ingressRules {
+		ingressRule.podIPs, err = c.convertANPPeersToIPSet(ingressRule.peers,
+			ingressRule.gressIndex, ingressRule.gressPrefix, anp.name)
+		if err != nil {
+			return fmt.Errorf("unable to create address set for "+
+				" rule %s with priority %d: %w", ingressRule.name, ingressRule.priority, err)
+		}
+	}
+	for _, egressRule := range anp.egressRules {
+		egressRule.podIPs, err = c.convertANPPeersToIPSet(egressRule.peers,
+			egressRule.gressIndex, egressRule.gressPrefix, anp.name)
+		if err != nil {
+			return fmt.Errorf("unable to create address set for "+
+				" rule %s with priority %d: %w", egressRule.name, egressRule.priority, err)
+		}
+	}
+
+	return nil
+}
+
+// convertANPPeersToIPSet creates a set of podIPs for all the peers passed as argument
+// This function also takes care of populating the adminNetworkPolicyPeer.namespaces cache
+// It also adds up all the podIPs that are supposed to be present in the created AS and returns them on
+// a per-rule basis so that the actual ops to transact these into the AS can be constructed using that
+func (c *Controller) convertANPPeersToIPSet(peers []*adminNetworkPolicyPeer, gressIndex int32, gressPrefix,
```

Comment:
you are not using `gressIndex`, `gressPrefix` or `anpName` method arguments for anything?

Reply:
oh must be a refactor glitch.. :( let me check

Reply:
fixed it, was a refactor/rebase leftover